weave

```diff
@@ -1382,13 +1382,17 @@ check_overlap() {
 # Claim addresses for a container in IPAM.  Expects to be called from
 # with_container_addresses.
 ipam_reclaim() {
-    for CIDR in $3 ; do
-        http_call $HTTP_ADDR PUT "/ip/$1/$CIDR?noErrorOnUnknown=true&?check-alive=true" || true
+    CONTAINER_ID="_"
+    if [ "$2" = "$CONTAINER_IFNAME" ] ; then
+        CONTAINER_ID="$1"
+    fi
```

Comment:
To me a) this deserves a comment, b) is "the wrong way round" (i.e. the `_` is the special case), and c) should be done with a `[...] || CONTAINER_ID="_"` expression.